plugins/base/frontend/src/main/components/utils/hotkey.ts

```diff
@@ -0,0 +1,53 @@
+import {detectOsKind, OsKind} from "./os";
+
+type ModifierKey = {
+    name: string
+    keyArg: string
+}
+
+class ModifierKeys {
+    static metaKey: ModifierKey = {name: "Command", keyArg: "Meta"}
+    static ctrlKey: ModifierKey = {name: "Ctrl", keyArg: "Control"}
+    static altKey: ModifierKey = {name: "Alt", keyArg: "Alt"}
+    static shiftKey: ModifierKey = {name: "Shift", keyArg: "Shift"}
+}
+
+const setOfKeys = [ModifierKeys.altKey, ModifierKeys.shiftKey, ModifierKeys.ctrlKey, ModifierKeys.metaKey]
+
+export class Hotkey {
+    private readonly osKind: OsKind;
+
+    constructor() {
+        this.osKind = detectOsKind()
+    }
+
+    public getOsAccelKeyName() {
+        return this.getOsAccelKey().name
+    }
+
+    public registerHotkeyWithAccel = (event: () => void, letter: string) => {
+        const osMetaKey = this.getOsAccelKey()
+        document.onkeydown = (keyDownEvent) => {
+            const isMetaKeyPressed = keyDownEvent.getModifierState(osMetaKey.keyArg)
+            const isOtherModifierKeyPressed = setOfKeys
+                .filter(key => key !== osMetaKey)
+                .map((otherKeys: ModifierKey) => keyDownEvent.getModifierState(otherKeys.keyArg))
+                .some(value => value)
+
+            if (isMetaKeyPressed && !isOtherModifierKeyPressed && keyDownEvent.key === letter) {
```

Comment:
I'd leave a comment here like you did in the PR comments - it might not be obvious at first sight. Something like

> We don't want to trigger action for hotkeys like Ctrl+K if is a subset of another key combination (e.g. Ctrl + Shift + K)